enterprise_access/apps/content_assignments/api.py

```diff
@@ -996,6 +997,14 @@ def expire_assignment(
         if automatic_expiration_reason == AssignmentAutomaticExpiredReason.NINETY_DAYS_PASSED:
             assignment.clear_pii()
 
+        try:
+            credit_request = assignment.credit_request
+            logger.info('Modifying credit request %s to expired', credit_request.uuid)
+            credit_request.state = SubsidyRequestStates.EXPIRED
+            credit_request.save()
+        except LearnerContentAssignment.credit_request.RelatedObjectDoesNotExist:  # pylint: disable=no-member
+            logger.info('Assignment %s has no credit request', assignment.uuid)
+
         assignment.save()
         send_assignment_automatically_expired_email.delay(assignment.uuid)
```

Comment:
For assignments that have a 1:1 mapping with a request, we should trigger our own email for the expired request instead of assignment to avoid confusion. The email content and functionality for handling expired requests doesn’t exist yet, so as an interim solution, we can simply add a condition to skip sending the assignment expiration email if an associated request is found.

```python
if not credit_request:
    send_assignment_automatically_expired_email.delay(assignment.uuid)